lib/jekyll/admin/server/static_file.rb

```diff
@@ -40,7 +47,16 @@ def static_files
         end
 
         def static_file
-          static_files.find { |f| f.path == static_file_path }
+          file = static_files.find { |f| f.path == static_file_path }
+          if not file
+            static_files.select do |f|
+              # Files that are in this directory
+              # Joined with / to ensure user can't do partial paths
+              f.path.start_with? File.join(static_file_path, "/")
```

Comment:
It currently ignores folders as an entity and instead presents a full recursive tree listing of files from the folder specified on down.